components/eamxx/src/share/field/field.cpp

```diff
@@ -127,11 +167,31 @@ get_component (const int i, const bool dynamic) {
   EKAT_REQUIRE_MSG (i>=0 && i<layout.dim(idim),
       "Error! Component index out of bounds [0," + std::to_string(layout.dim(idim)) + ").\n");
 
-  // Add _$idim to the field name, to avoid issues if the subfield is stored
+  // Add _$i to the field name, to avoid issues if the subfield is stored
   // in some structure that requires unique names (e.g., a remapper)
   return subfield (fname + "_" + std::to_string(i),idim,i,dynamic);
 }
 
+Field Field::get_components(const int i1, const int i2) {
```

Comment:
I would change the input vars names, since i1/i2 do not really explain what happens. One may misunderstand them for "oh, i only get 2 slices, i1 and i2".  I would stick with beg/end, which are more self-documenting.